examples/chef/common/stubs.cpp

```diff
@@ -8,6 +8,159 @@
+        char name[kMaxNameLength];
+        DlCredential credentialMap[kNumCredentialsPerUser];
+    };
+
+    struct Endpoint
+    {
+        chip::EndpointId id;
+        User users[kNumUsersPerEndpoint];
+        Credential credentials[kNumCredentialsPerEndpoint];
+    };
+
+    static LockManager & Instance()
+    {
+        static LockManager instance;
+        return instance;
+    }
+
+    LockManager() { defaultInitialize(); }
+
+    bool getUser(chip::EndpointId endpointId, uint16_t userIndex, EmberAfPluginDoorLockUserInfo & user)
+    {
+        auto ep = findEndpoint(endpointId);
+        if (!ep)
+            return false;
+        if (userIndex >= kNumUsersPerEndpoint)
+            return false;
+        user = ep->users[userIndex].info;
+        return true;
+    }
+
+    bool setUser(chip::EndpointId endpointId, uint16_t userIndex, chip::FabricIndex creator, chip::FabricIndex modifier,
+                 const chip::CharSpan & userName, uint32_t uniqueId, DlUserStatus userStatus, DlUserType usertype,
+                 DlCredentialRule credentialRule, const DlCredential * credentials, size_t totalCredentials)
+    {
+        auto ep = findEndpoint(endpointId);
+        if (!ep)
+            return false;
+        if (userIndex >= kNumUsersPerEndpoint || totalCredentials > kNumCredentialsPerUser)
+            return false;
+        ep->users[userIndex].set(userName, uniqueId, userStatus, usertype, credentialRule);
+        ep->users[userIndex].info.creationSource     = DlAssetSource::kMatterIM;
+        ep->users[userIndex].info.createdBy          = creator;
+        ep->users[userIndex].info.modificationSource = DlAssetSource::kMatterIM;
+        ep->users[userIndex].info.lastModifiedBy     = modifier;
+        for (size_t i = 0; i < totalCredentials; i++)
+            ep->users[userIndex].addCredential(credentials[i].CredentialType, credentials[i].CredentialIndex);
+        return true;
+    }
+
+    bool getCredential(chip::EndpointId endpointId, uint16_t credentialIndex, DlCredentialType credentialType,
+                       EmberAfPluginDoorLockCredentialInfo & credential)
+    {
+        auto ep = findEndpoint(endpointId);
+        if (!ep)
+            return false;
+        if (credentialIndex >= kNumCredentialsPerEndpoint)
+            return false;
+        credential = ep->credentials[credentialIndex].info;
+        return true;
+    }
+
+    bool setCredential(chip::EndpointId endpointId, uint16_t credentialIndex, chip::FabricIndex creator, chip::FabricIndex modifier,
+                       DlCredentialStatus credentialStatus, DlCredentialType credentialType, const chip::ByteSpan & credentialData)
+    {
+        auto ep = findEndpoint(endpointId);
+        if (!ep)
+            return false;
+        if (credentialIndex >= kNumCredentialsPerEndpoint)
+            return false;
+        if (!ep->credentials[credentialIndex].set(credentialStatus, credentialType, credentialData))
```

Comment:
This does not look right.  credentialIndex is per-type, not a global index, in the spec.

Reply:
Ah yeah I was wondering since returning false if it didn't match didn't work. Changed it to be a 2d array